core/ops_builtin.rs

```diff
@@ -168,6 +169,26 @@ async fn op_read(
   resource.read_return(buf).await.map(|(n, _)| n as u32)
 }
 
+#[op]
+async fn op_read_all(
```

Comment:
Nice. #16115 will allow us to get rid of multiple allocations in this op in a follow up.